charts/opentelemetry-demo/values.yaml

```diff
@@ -286,6 +286,8 @@ components:
         port: 8081
       initialDelaySeconds: 30
       periodSeconds: 10
+    initContainers:
+      dependency: '{{ include "otel-demo.name" . }}-ffspostgres 5432'
```

Comment:
Is there a use case to make it configurable? Maybe just put this value right in the template?

Reply:
Not sure what you mean -- there's a use case for other containers to init (for example, perhaps things that need redis should wait for redis), but this is kind of a workaround in general that I'd prefer we deprecate eventually.

Reply:
Instead of `initContainer` being an object that allows defining a dependency, can we instead let the initContainer object be all the parts of the initContainer?  This is how the collector chart works.  In _objects.tpl we can then do something like:

```yaml
{{- if .Values.initContainers }}
initContainers:
  {{- tpl (toYaml .Values.initContainers) . | nindent 2 }}
{{- end }}
```

For the values.yaml we'd update the comment at the start of the services detailing how initContainer can be used and then for featureflagservice we'd supply the default init container that it needs.

This gives us the container we need and also the flexibility for users to add any sort of initContainer they want.  What do you think?